### PR TITLE
Add rector

### PR DESCRIPTION
**Pull request by juliushaertl**
Add rector and apply automatic fixes :)

### CODE REVIEW

File: lib/Backgroundjobs/Cleanup.php [hunk 1]
```diff
@@ -13,15 +13,8 @@
 use OCP\IDBConnection;
 
 class Cleanup extends TimedJob {
-	/** @var IDBConnection */
-	private $db;
-	/** @var WopiMapper $wopiMapper */
-	private $wopiMapper;
-
-	public function __construct(ITimeFactory $time, IDBConnection $db, WopiMapper $wopiMapper) {
+	public function __construct(ITimeFactory $time, private IDBConnection $db, private WopiMapper $wopiMapper) {
```
Comment (blizzz) [inline lib/Backgroundjobs/Cleanup.php:16]: those should go then one parameter per line.

File: lib/Controller/DocumentAPIController.php [hunk 2]
```diff
@@ -29,23 +29,8 @@
...
-	private $rootFolder;
-	private $shareManager;
-	private $templateManager;
-	private $l10n;
-	private $logger;
-	private $lockManager;
-	private $userId;
-
-	public function __construct(IRequest $request, IRootFolder $rootFolder, IManager $shareManager, TemplateManager $templateManager, IL10N $l10n, LoggerInterface $logger, ILockManager $lockManager, $userId) {
+	public function __construct(IRequest $request, private IRootFolder $rootFolder, private IManager $shareManager, private TemplateManager $templateManager, private IL10N $l10n, private LoggerInterface $logger, private ILockManager $lockManager, private $userId) {
```
Comment (blizzz) [inline lib/Controller/DocumentAPIController.php:32]: Also split to one line per parameter

File: lib/Db/AssetMapper.php [hunk 3]
```diff
@@ -17,14 +17,8 @@ class AssetMapper extends QBMapper {
 	/** @var int Lifetime of a token is 10 minutes */
 	public const TOKEN_TTL = 600;
 
-	private ISecureRandom $random;
-	private ITimeFactory $time;
-
-	public function __construct(IDBConnection $db, ISecureRandom $random, ITimeFactory $timeFactory) {
+	public function __construct(IDBConnection $db, private ISecureRandom $random, private ITimeFactory $time) {
```
Comment (blizzz) [inline lib/Db/AssetMapper.php:20]: and here

File: lib/Events/DocumentOpenedEvent.php [hunk 4]
```diff
@@ -11,12 +11,7 @@
 use OCP\Files\Node;
 
 class DocumentOpenedEvent extends \OCP\EventDispatcher\Event {
-	private ?string $userId;
-	private Node $node;
-
-	public function __construct(?string $userId, Node $node) {
-		$this->userId = $userId;
-		$this->node = $node;
+	public function __construct(private ?string $userId, private Node $node) {
```
Comment (blizzz) [inline lib/Events/DocumentOpenedEvent.php:14]: now this short, but would also be better split up

File: lib/Listener/BeforeFetchPreviewListener.php [hunk 5]
```diff
@@ -24,16 +24,7 @@
...
-	private IUserSession $userSession;
-	private IRequest $request;
-	private IManager $shareManager;
-
-	public function __construct(PermissionManager $permissionManager, IUserSession $userSession, IRequest $request, IManager $shareManager) {
-		$this->permissionManager = $permissionManager;
-		$this->userSession = $userSession;
-		$this->request = $request;
-		$this->shareManager = $shareManager;
+	public function __construct(private PermissionManager $permissionManager, private IUserSession $userSession, private IRequest $request, private IManager $shareManager) {
```
Comment (blizzz) [inline lib/Listener/BeforeFetchPreviewListener.php:27]: and here

File: lib/Listener/LoadViewerListener.php [hunk 6]
```diff
@@ -20,17 +20,7 @@
...
-	/** @var InitialStateService */
-	private $initialStateService;
-
-	private ?string $userId = null;
-
-	public function __construct(PermissionManager $permissionManager, InitialStateService $initialStateService, ?string $userId) {
-		$this->permissionManager = $permissionManager;
-		$this->initialStateService = $initialStateService;
-		$this->userId = $userId;
+	public function __construct(private PermissionManager $permissionManager, private InitialStateService $initialStateService, private ?string $userId) {
```
Comment (blizzz) [inline lib/Listener/LoadViewerListener.php:23]: and here

File: lib/Listener/ShareLinkListener.php [hunk 7]
```diff
@@ -21,14 +21,7 @@
...
 class ShareLinkListener implements \OCP\EventDispatcher\IEventListener {
-	/** @var PermissionManager */
-	private $permissionManager;
-	/** @var InitialStateService */
-	private $initialStateService;
-
-	public function __construct(PermissionManager $permissionManager, InitialStateService $initialStateService) {
-		$this->permissionManager = $permissionManager;
-		$this->initialStateService = $initialStateService;
+	public function __construct(private PermissionManager $permissionManager, private InitialStateService $initialStateService) {
```
Comment (blizzz) [inline lib/Listener/ShareLinkListener.php:24]: and here

File: lib/Service/DemoService.php [hunk 8]
```diff
@@ -10,18 +10,7 @@
...
-	private $cache;
-	/**
-	 * @var IClientService
-	 */
-	private $clientService;
-
-	public function __construct(ICache $cache, IClientService $clientService) {
-		$this->cache = $cache;
-		$this->clientService = $clientService;
+	public function __construct(private ICache $cache, private IClientService $clientService) {
```
Comment (blizzz) [inline lib/Service/DemoService.php:13]: and here

File: lib/Service/FederationService.php [hunk 9]
```diff
@@ -32,34 +32,14 @@
...
-	/** @var TokenManager */
-	private $tokenManager;
-	/** @var IRequest */
-	private $request;
-	/** @var IURLGenerator */
-	private $urlGenerator;
-
-	public function __construct(ICacheFactory $cacheFactory, IClientService $clientService, LoggerInterface $logger, TokenManager $tokenManager, AppConfig $appConfig, IRequest $request, IURLGenerator $urlGenerator) {
+
+	public function __construct(ICacheFactory $cacheFactory, private IClientService $clientService, private LoggerInterface $logger, private TokenManager $tokenManager, private AppConfig $appConfig, private IRequest $request, private IURLGenerator $urlGenerator) {
```
Comment (blizzz) [inline lib/Service/FederationService.php:38]: and here

File: lib/Settings/Section.php [hunk 10]
```diff
@@ -12,17 +12,7 @@
...
-	/** @var CapabilitiesService */
-	private $capabilitites;
-	/** @var IL10N */
-	private $l10n;
-
-	public function __construct(IURLGenerator $url, CapabilitiesService $capabilities, IL10N $l10n) {
-		$this->url = $url;
-		$this->capabilitites = $capabilities;
-		$this->l10n = $l10n;
+	public function __construct(private IURLGenerator $url, private CapabilitiesService $capabilitites, private IL10N $l10n) {
```
Comment (blizzz) [inline lib/Settings/Section.php:15]: and here

**Review by blizzz — COMMENTED** [review]
Nitpicking on some constructor parameter lists

File: lib/Backgroundjobs/Cleanup.php (hunk 1 above)
Comment (juliushaertl) [reply lib/Backgroundjobs/Cleanup.php]: Applied with the new php-cs-fixer config as it has https://github.com/kubawerlos/php-cs-fixer-custom-fixers#MultilinePromotedPropertiesFixer included

**Review by blizzz — APPROVED** [review]

**Pull request merged**